### ISSUE TITLE
Reconfigure How Inventory is Handled for Openshift-Ansible

### ISSUE THREAD

**Issue by stevekuznetsov** [stevekuznetsov/origin-ci-tool#18]
There are generally three independent parts of the inventory necessary for a full install:
 - the static definition of groupings like:

    ```ini
    [OSEv3:children]
    masters
    nodes
    ```
 - the dynamic mapping of machines to these groups
 - the user-provided installation config

Currently, these are vaguely mushed together into the inventory file. The correct approach would be to co-locate these scripts in a structure like this:
```
inventory/
├── dynamic_inventory_config.ini # API keys, etc
├── dynamic_inventory.py         # Script to get host information from an external API
└── static_grouping.ini          # Rigid group structure openshift-ansible requires
group_vars/
└── OSEv3.yaml                   # User-provided configuration options for the install
```

We would let the specific provisioner choose the dynamic inventory script and configure it, internalizing mappings like [these](https://github.com/openshift/openshift-ansible/blob/master/playbooks/byo/openshift-cluster/cluster_hosts.yml), let the `openshift-ansible` repo dictate the rigid, static grouping, and let the user bring their configuration file.

**Comment by stevekuznetsov** [comment]
@abutcher has confirmed that for the time being a global dump of config in `[OSEv3:vars]` will be fine, and since we only care about supporting a simple, opinionated entrypoint for the underlying tools, we are OK just exposing to the user a flag along the lines of "where is your config file".

@jdetiber FYI this seems like a reasonable separation of concerns re: our conversation earlier.

@sallyom I was thinking about how you are currently submoduling the entirety of OpenShift-Ansible and how much information you had to bring to the table in order to get the Online install to function as you'd like. In a perfect world, all you would need is the configuration YAML. This would let you be agnostic to whatever tool was used to provision the hosts you are using and furthermore decouple the relationship between you and the BYO playbook a little bit, but nevertheless version skew would be an issue.  If, however, we were able to get such a configuration file to exist in your repo, I could pretty easily support a "Online-on-top" type of install where I controlled the provisioning by using the config file and running the BYO playbook myself.
